Source/Module/Login/LoginViewModel.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  LoginViewModel.swift
+//  GithubNotifications
+//
+//  Created by Chananchida F. on 9/13/21.
+//  Copyright © 2021 Nimblehq. All rights reserved.
+//
+
+import Foundation
+import SwiftUI
+import Combine
+
+protocol LoginViewModelProtocol {
+
+    func getUser(token: String, completion: @escaping CompletionHandler)
+}
+
+final class LoginViewModel: ObservableObject, LoginViewModelProtocol {
+
+    private var userService: UserServiceProtocol
+    private var userSession: UserSessionStorageProtocol
+
+    var userSubscription: AnyCancellable?
+
+    init(userService: UserServiceProtocol, userSession: UserSessionStorageProtocol) {
+        self.userService = userService
+        self.userSession = userSession
+    }
+    
+    func getUser(token: String, completion: @escaping CompletionHandler) {
+        userSession.authToken = token
+        userSubscription = userService.getUser()
+            .sink(
+                receiveCompletion: { completion in
+                    #warning("Show error when failed to get user")
+                }, receiveValue: { [weak self] user in
+                    self?.userSession.username = user.login
+                    completion()
+                }
+            )
```

Comment:
feel free to play around with [assign(to:)](https://developer.apple.com/documentation/combine/fail/assign(to:)) not sure if it will work here, think of this as a demo project where we are less strict about format but encourage experiment. Example
```
userService.getUser()
            .assertNoFailure()
            .compactMap { $0.login }
            .assign(to: \.username, on: userSession)
            .store(in: &cancellables)
```

Reply:
@blyscuit Thank you, will check it out 😄